src/Components/Components/src/Forms/InputComponents/InputNumber.cs

```diff
@@ -62,19 +40,19 @@ static InputNumber()
         protected override void BuildRenderTree(RenderTreeBuilder builder)
         {
             builder.OpenElement(0, "input");
-            builder.AddAttribute(1, "step", _stepAttributeValue); // Before the splat so the user can override
+            builder.AddAttribute(1, "step", _stepAttributeValue);
             builder.AddMultipleAttributes(2, AdditionalAttributes);
             builder.AddAttribute(3, "type", "number");
             builder.AddAttribute(4, "class", CssClass);
-            builder.AddAttribute(5, "value", BindMethods.GetValue(CurrentValueAsString));
+            builder.AddAttribute(5, "value", BindConverter.FormatValue(CurrentValueAsString));
             builder.AddAttribute(6, "onchange", EventCallback.Factory.CreateBinder<string>(this, __value => CurrentValueAsString = __value, CurrentValueAsString));
             builder.CloseElement();
         }
 
         /// <inheritdoc />
         protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage)
         {
-            if (_parser(value, out result))
+            if (BindConverter.TryConvertTo<T>(value, CultureInfo.InvariantCulture, out result))
```

Comment:
This seemed like a good tradeoff, it adds a dictionary lookup, but removes an instance of boxing. 

The same optimization doesn't apply to dates, because of the format string.